### PR TITLE
filter

### PR DESCRIPTION
**Pull request by arssivka**
Add message filter and some changes in root node, topic and billboard classes.

### CODE REVIEW

**Comment by arssivka** [comment]
[Фильтры для сообщений в топиках](https://trello.com/c/AD286z07/23--)

File: include/rrc/core/RootNode.h [hunk 1]
```diff
@@ -90,10 +90,16 @@ namespace rrc {
...
          * @return Set of topic names
          */
-        std::set<Key> getTopicNames() const {
+        std::vector<Key> getTopicNames() const {
             return mBillboard.getKeys();
         }
 
+        void removeTopic(const Key& topicName);
+
+        void setTopicAutoRemoveFlag(const Key& topicName, bool);
```
Comment (TekatoD) [inline include/rrc/core/RootNode.h]: There is only "bool" in the parameters

File: include/rrc/core/RootNode.h [hunk 2]
```diff
@@ -90,10 +90,16 @@ namespace rrc {
          * @brief Returns set of avaliable topic names
          * @return Set of topic names
          */
-        std::set<Key> getTopicNames() const {
+        std::vector<Key> getTopicNames() const {
             return mBillboard.getKeys();
         }
 
+        void removeTopic(const Key& topicName);
```
Comment (TekatoD) [inline include/rrc/core/RootNode.h]: Test this.

File: include/rrc/core/RootNode.h (hunk 1 above)
Comment (TekatoD) [inline include/rrc/core/RootNode.h]: And test this.

File: include/rrc/core/RootNode.h [hunk 3]
```diff
@@ -90,10 +90,16 @@ namespace rrc {
...
+
+        void setTopicAutoRemoveFlag(const Key& topicName, bool);
+
+        void setTopicMessageFilter(const Key& topicName, AbstractMessageFilterPtr filter);
```
Comment (TekatoD) [inline include/rrc/core/RootNode.h]: Also test this.

File: include/rrc/core/RootNode.h [hunk 4]
```diff
@@ -90,10 +90,16 @@ namespace rrc {
          * @brief Returns set of avaliable topic names
          * @return Set of topic names
          */
-        std::set<Key> getTopicNames() const {
+        std::vector<Key> getTopicNames() const {
```
Comment (TekatoD) [inline include/rrc/core/RootNode.h]: Test this.

File: include/rrc/core/RootNode.h (hunk 1 above)
Comment (arssivka) [inline include/rrc/core/RootNode.h]: Done!

File: include/rrc/core/RootNode.h (hunk 4 above)
Comment (arssivka) [inline include/rrc/core/RootNode.h]: Okay(

File: include/rrc/core/RootNode.h (hunk 2 above)
Comment (arssivka) [inline include/rrc/core/RootNode.h]: I'm a cool tester! I have tested this!

**Pull request merged**